cognite/neat/core/rules/exporter/rules2graphql.py

```diff
@@ -1,21 +1,99 @@
+
+from cognite.neat.core.rules.analysis import get_class_property_pairs
+from cognite.neat.core.rules.models import DATA_TYPE_MAPPING, TransformationRules
 from cognite.neat.core.utils.utils import generate_exception_report
 
 
+TYPE = (
+    "{% include 'type_header' %}type {{ class_definition.class_id }} {{'{'}}"
+    "{%-for property_definition in class_properties%}"
+    "{% include 'field' %}"
+    "{% endfor %}\n}\n"
+)
+
+TYPE_HEADER = (
+    "{%- if header %}"
+    "{%- if class_definition.description and class_definition.class_name %}"
+    '"""\n{{class_definition.description}}\n@name {{ class_definition.class_name }}\n"""\n{##}\n'
+    "{%- elif class_definition.description %}"
+    '\n"""\n{{class_definition.description}}\n"""\n'
+    "{%- endif %}"
+    "{%- endif %}"
+)
+
+
+FIELD = (
+    "{% include 'field_header' %}\n"
+    "  {{ property_definition.property_id }}: "
+    "{%-if property_definition.property_type == 'DatatypeProperty'%}"
+    "{% include 'attribute_occurrence' %}"
+    "{%-else%}"
+    "{% include 'edge_occurrence' %}"
+    "{%- endif -%}"
+)
+
+FIELD_HEADER = (
+    "{%- if header %}"
+    "{%- if property_definition.description and property_definition.property_name %}"
+    '\n  """\n  {{property_definition.description}}'
+    "\n  @name {{ property_definition.property_name }}"
+    '\n  """\n  '
+    "{%- elif property_definition.description %}"
+    '\n  """\n  {{property_definition.description}}'
+    '\n  """\n  '
+    "{%- endif %}"
+    "{%- endif %}"
+)
+
+ATTRIBUTE_OCCURRENCE = (
+    "{%-if property_definition.min_count and property_definition.max_count == 1%}"
+    " {% include 'value_type' %}!"
+    "{%-elif property_definition.min_count and property_definition.max_count != 1%}"
+    " [{% include 'value_type' %}!]!"
+    "{%-elif property_definition.max_count != 1%}"
+    " [{% include 'value_type' %}!]!"
+    "{%-else%}{% include 'value_type' %}"
+    "{%- endif -%}"
+)
+
+EDGE_OCCURRENCE = (
+    "{%-if not(property_definition.min_count and property_definition.max_count == 1)%}"
+    " [{% include 'value_type' %}]"
+    "{%-else%}"
+    " {% include 'value_type' %}"
+    "{%- endif -%}"
+)
+
+FIELD_VALUE_TYPE = """{{value_type_mapping[property_definition.expected_value_type]['GraphQL']
+                if property_definition.expected_value_type in value_type_mapping
+                else property_definition.expected_value_type}}"""
+
+
+rules2graphql_template = Environment(
+    loader=DictLoader(
+        {
+            "type_header": TYPE_HEADER,
+            "type": TYPE,
+            "field_header": FIELD_HEADER,
+            "field": FIELD,
+            "value_type": FIELD_VALUE_TYPE,
+            "edge_occurrence": EDGE_OCCURRENCE,
+            "attribute_occurrence": ATTRIBUTE_OCCURRENCE,
+        }
+    ),
+    cache_size=1000,
+).get_template("type")
+
+
 def rules2graphql_schema(
     transformation_rules: TransformationRules,
+    header: bool = False,
```

Comment:
This is missing from the docstring.

Reply:
now this is field `verbose` under `GraphQLSchema` class